src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -81,6 +113,15 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
 	common.SetEnvironmentVariableInScript(scriptBuilder, "PORT", gen.BindPort, DefaultBindPort)
 	scriptBuilder.WriteString("export ASPNETCORE_URLS=http://*:$PORT\n\n")
 
+	logger.LogInformation("Setting up Kestrel Endpoints with BindPort and BindPort2 Env variables if .NET runtime version >= 7")
+	if gen.isDotnetRuntimeVersionMeetConstraint("7.0.0-0") {
+		scriptBuilder.WriteString("if [ -z \"$" + gen.BindPort2 + "\" ]; then" + "\n")
```

Comment:
We should also validate that BindPort is set as well, as we'll need to read its value to set Kestrel__Endpoints__Http1__Url

Reply:
Or the $PORT value if it's easier, as it's already defined above?

Reply:
The $PORT value was setup here, and it has default value of 8080:
`	common.SetEnvironmentVariableInScript(scriptBuilder, "PORT", gen.BindPort, DefaultBindPort)
`

Reply:
OK fair enough! Thanks for clarifying